examples/gallery/embellishments/inset_rectangle_region.py

```diff
@@ -0,0 +1,48 @@
+"""
+Inset map showing a rectangle region
+------------------------------------
+
+The :meth:`pygmt.Figure.inset` method adds an inset figure inside a larger
+figure. The function is called using a ``with`` statement, and its position,
+box, offset, and margin can be customized. Plotting methods called within the
+``with`` statement plot into the inset figure.
+"""
+
+import numpy as np
+import pygmt
+
+# Set the region of the main figure
+region = [137.5, 141, 34, 37]
+
+fig = pygmt.Figure()
+
+# Plot the base map of the main figure
+fig.basemap(region=region, projection="M12c", frame=["WSne", "af"])
+
+# Set the land color to "lightbrown", the water color to "azure1", the shorelines
+# width to "2p", the area threshold to 1000 km^2 for the main figure
+fig.coast(land="lightbrown", water="azure1", shorelines="2p", area_thresh=1000)
+
+# Create an inset map, setting the position to bottom right, the width to
+# 3 centimeters, the height to 3.6 centimeters, and the x- and y-offsets to
+# 0.1 centimeters. Draws a rectangular box around the inset with a fill color
+# of "white" and a pen of "1p".
+with fig.inset(position="jBR+w3c/3.6c+o0.1c", box="+gwhite+p1p"):
+    # Plot the Japan main land in the inset using coast. "M?" means Mercator
```

Comment:
If it's not too much work, would it be possible to change to a different projection system besides Mercator? E.g. Universal Transverse Mercator (U)? I worry about all the people who are going to use Mercator for their publications because it was a default copied from some gallery example!

Reply:
> If it's not too much work, would it be possible to change to a different projection system besides Mercator? E.g. Universal Transverse Mercator (U)? I worry about all the people who are going to use Mercator for their publications because it was a default copied from some gallery example!

I think it's really a good suggestion. Maybe we should add a tutorial to discuss how to choose projection. But I am not very familiar with the difference between different projections.

I test [`U`](https://docs.generic-mapping-tools.org/latest/cookbook/map-projections.html#universal-transverse-mercator-utm-projection-ju-ju), the figure looks a little oblique. Ping @GenericMappingTools/python @GenericMappingTools/python-contributors for any suggestion?
```Python
fig.basemap(region=region, projection="U54S/12c", frame=["WSne", "af"])
...
# Inset
    fig.coast(
        region=[129, 146, 30, 46],
        projection="U54S/?",
        ...
    )
```
![Screenshot from 2021-03-12 15-06-46](https://user-images.githubusercontent.com/50591376/110904878-9a9e1c00-8344-11eb-997d-10ea4d2d9b82.png)

Reply:
Now there are several examples with different projections:
- [Political Boundaries](https://www.pygmt.org/dev/gallery/maps/borders.html#sphx-glr-gallery-maps-borders-py): `I`
- [Color land and water](https://www.pygmt.org/dev/gallery/maps/land_and_water.html#sphx-glr-gallery-maps-land-and-water-py): `W`
- [Shorelines](https://www.pygmt.org/dev/gallery/maps/shorelines.html#sphx-glr-gallery-maps-shorelines-py): `W`
- [Sampling along tracks](https://www.pygmt.org/dev/gallery/images/track_sampling.html#sphx-glr-gallery-images-track-sampling-py): `Cyl_stere`
- [Focal mechanisms](https://www.pygmt.org/dev/gallery/seismology/meca.html#sphx-glr-gallery-seismology-meca-py): `M`
- [Inset](https://www.pygmt.org/dev/gallery/embellishments/inset.html#sphx-glr-gallery-embellishments-inset-py): Default projection `Q15c+`

Reply:
> I think it's really a good suggestion. Maybe we should add a tutorial to discuss how to choose projection. But I am not very familiar with the difference between different projections.

> I test [`U`](https://docs.generic-mapping-tools.org/latest/cookbook/map-projections.html#universal-transverse-mercator-utm-projection-ju-ju), the figure looks a little oblique. Ping @GenericMappingTools/python @GenericMappingTools/python-contributors for any suggestion?

Not an expert with GMT projection codes either, but I think it's ok to have it a little oblique. Maybe take a look at https://docs.generic-mapping-tools.org/6.1/gallery/ex28.html which managed a square-ish output somehow.

P.S. Please try to keep pinging of teams to a minimal (e.g. just one team if possible instead of 2 or 3). Not everyone checks their notification area all the time, and it's best not to spam people's notification area with dozens of mentions if they decide to go away for the weekend.

Reply:
> P.S. Please try to keep pinging of teams to a minimal (e.g. just one team if possible instead of 2 or 3). Not everyone checks their notification area all the time, and it's best not to spam people's notification area with dozens of mentions if they decide to go away for the weekend.

Okay~ Thanks for the suggestion.

Reply:
Personally I would 1) just try and solve it myself first, 2) ask a question without pinging anyone (it's ok to leave a PR like that for a few days, those who are interested will usually answer) and then 3) ping someone from the team if you get no reply. I realize it's getting close to the release date and we all want our PRs to be merged ASAP, but please try and be considerate of people's inboxes too!

That said, I think the slightly oblique Japan map on a UTM projection is fine, it actually looks a bit cool to be honest! After trying for a bit, I think (though not 100%) that the only way to get a rectangular/non-oblique map is to plot a grid first in linear projection system (e.g. -Jx1:250000) which would require knowing the region bounds in UTM unit metres rather than lon/lat. This will be a bit too complicated for a gallery example, so let's just keep it easy and use `U54S/12c`.

Reply:
> Personally I would 1) just try and solve it myself first, 2) ask a question without pinging anyone (it's ok to leave a PR like that for a few days, those who are interested will usually answer) and then 3) ping someone from the team if you get no reply. I realize it's getting close to the release date and we all want our PRs to be merged ASAP, but please try and be considerate of people's inboxes too!

Great~

> That said, I think the slightly oblique Japan map on a UTM projection is fine, it actually looks a bit cool to be honest! After trying for a bit, I think (though not 100%) that the only way to get a rectangular/non-oblique map is to plot a grid first in linear projection system (e.g. -Jx1:250000) which would require knowing the region bounds in UTM unit metres rather than lon/lat. This will be a bit too complicated for a gallery example, so let's just keep it easy and use `U54S/12c`.

I would like to keep `M` for the following reasons: 1) If we assume that PyGMT users will copy the code to use for their own figures, I think they usually will plot symbols/data/lines/... in the figure, too. If we use `U` instead of `M`, they will also have to face the issue of how to plot those things in a UTM map. 2) We may have a tutorial about how to use projection in the future and GMT users should know how to choose a better projection themselves.

Reply:
> I would like to keep `M` for the following reasons: 1) If we assume that PyGMT users will copy the code to use for their own figures, I think they usually will plot symbols/data/lines/... in the figure, too. If we use `U` instead of `M`, they will also have to face the issue of how to plot those things in a UTM map.

I'm pretty sure users can plot symbols/data/lines on a UTM map just as usual compared to a Mercator map right? I.e. use the same georeferenced data in longitude/latitude. The projection only distorts the shape of the Earth, not the data points or how they can be plotted.

> 2) We may have a tutorial about how to use projection in the future and GMT users should know how to choose a better projection themselves.

Yes I agree that a tutorial for projections is good (though this is a big geography lesson in itself).

Reply:
> > I would like to keep `M` for the following reasons: 1) If we assume that PyGMT users will copy the code to use for their own figures, I think they usually will plot symbols/data/lines/... in the figure, too. If we use `U` instead of `M`, they will also have to face the issue of how to plot those things in a UTM map.
> 
> I'm pretty sure users can plot symbols/data/lines on a UTM map just as usual compared to a Mercator map right? I.e. use the same georeferenced data in longitude/latitude. The projection only distorts the shape of the Earth, not the data points or how they can be plotted.

Good to know. It seems that my understanding is a little wrong. Done in 0ea30e2.